### DISCUSSION TITLE
How to upgrade/migrate ZBDongle-E from 6.x firmware to 7.4?

### DISCUSSION THREAD

**Discussion by Skirmisher** [Koenkk/zigbee2mqtt#22919] [Devices]
I'm trying my best to catch up with the whole deal with the changeover from the `ezsp` driver to the rewritten `ember` driver, but it's been throwing me for a loop, having to look in several different places and needing to determine what sources of info are up to date... (For instance, the website FAQ says that coordinator backup isn't supported for EZSP, so I didn't bother attempting it, but then the "adapters" page says it does...)

I have a ZBDongle-E that was running some 6.x firmware version that it came with (may have been 6.9 or 6.10), which has been running happily with Z2M (Home Assistant addon) for the past year. I successfully flashed [`ncp-uart-hw-v7.4.2.0-zbdonglee-115200.gbl`](https://github.com/darkxst/silabs-firmware-builder/blob/main/firmware_builds/zbdonglee/ncp-uart-hw-v7.4.2.0-zbdonglee-115200.gbl) via the Silicon Labs Flasher addon from Home Assistant. Then I changed over Z2M's config to use `adapter: ember` and tried to start it, but it complained about the backup file being unsupported (`Current backup file is from an unsupported EZSP version (min: 12).`). So I changed the driver back to `ezsp` thinking I might need it to do something first, but all it gave me was `Failure to init network`. Finally, I downgraded the adapter to [NCP 6.10.3 from itead's repo](https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/blob/master/Dongle-E/NCP/ncp-uart-sw_EZNet6.10.3_V1.0.1.gbl), restored the partial backup of Z2M I had made in Home Assistant, and tried starting Z2M again to restore functionality... but the device list is completely empty! Am I screwed?

Edit: Right, so the Home Assistant-side addon backup doesn't actually back up any of Z2M's own configuration. At the moment, in Z2M's config directory, I have a `coordinator_backup.json` with an empty `devices` array, a `configuration.yaml` that lists my device descriptions and config tweaks, a `database.db` with no devices in it, and a `database.db.backup` with all the devices in it. Seems like I'll have to re-pair everything... but I'm still not sure what I should have done to make this go smoothly.

**Comment by francisp2** [comment]
Upgrade from 6.x to 7.3 (version 12), run Zigbee2MQTT so it can make a backup, upgrade from 7.3 to 7.4 (then Zigbee2MQTT can restore the backup)

**Comment by TolianIPB** [comment]
I was successful in upgrading ZBDongle-E from 6.x firmware to 7.4.
My device is:
Sonoff ZBDongle-E V2
CH9102
EFR32MG21

The firmware update didn't work for me on the website.

I flashed the firmware:
https://github.com/darkxst/silabs-firmware-builder/blob/main/firmware_builds/zbdonglee/ncp-uart-hw-v7.4.2.0-zbdonglee-115200.gbl

Instructions for flshing for Linux i've got there:
https://www.zigbee2mqtt.io/devices/ZBDongle-E.html

I edited the file:
config/zigbee2mqtt/configuration.yaml
```
 - adapter: ezcp
 + adapter: ember
```

I deleted:
config/zigbee2mqtt/database.db.backup
config/zigbee2mqtt/coordinator_backup.json

And everything worked just great!
No settings were lost.

**Comment by sKinyo** [comment]
Upgrade from 6.10.3 directly to 7.4.2 does work. I have done it last week.
After the upgrade Zigbee2MQTT must be started with the ezsp setting otherwise you get the "Current backup file is from an unsupported EZSP version (min: 12)" error.

In a second step you can switch to ember.
No repairing needed, all devices available.

**Reply by Koenkk** [reply]
I believe that deleting the `coordinator_backup.json` from the `data` folder should also fix the issue.

**Comment by Manni51** [comment]
I'm still trying to flash the ZBDongle-E using a Win11 Laptop. I have tried "connect" on https://darkxst.github.io/silabs-firmware-builder/. This failed even if I have installed suggested driver.

Does anyone succeed flashing using Win11? Please let me know how to do.

**Reply by Manni51** [reply]
I'm sorry. 
After installing requested driver and reboot. The window which popups after "connect" was slightly different. The dongle is listed in the window and you need to "select" the dongle and continue!
The software suggests to install a 7.3 version. As zigbee2mqtt requests 7.4.x you need to select the previously downloaded 7.4.x gpl file,

**Comment by Sergiocastis** [comment]
Tengo el mismo problema estoy intentándolo por Windows, instalo los controladores me reconoce el dispositivo pero al darle a conectar me dice que fue imposible

**Reply by Manni51** [reply]
If you are running zigbee2mqtt on linux (i.e. raspi) you may flash as follows:
- put the firmware to linux (i.e. your home folder)
- stop zigbee2mqtt
- if not already done: pip install universal-silabs-flasher
- flash your dongle using universal-silabs-flasher with the new firmware
- start zigbee2mqtt
